packages/hebao_v1/contracts/base/MetaTxModule.sol

```diff
@@ -101,27 +101,39 @@ contract MetaTxModule is BaseModule
         uint startGas = gasleft();
         require(startGas >= gasSetting[2], "OUT_OF_GAS");
 
-        address wallet = extractWalletAddress(data);
+        // Copy data to memory
+        bytes memory _data = data;
+
+        address wallet = extractWalletAddress(_data);
         bytes32 metaTxHash = getSignHash(
             wallet, // from
             address(this),  // to. Note the relayer can only call its own methods.
             msg.value, // value
-            data,
+            _data,
             nonce,
             gasSetting
         );
 
-        address[] memory signers = extractMetaTxSigners(wallet, extractMethod(data), data);
-        metaTxHash.verifySignatures(signers, signatures);
+        bytes4 method = extractMethod(_data);
+        // If we exectute multiple transactions using `executeTransactions` we don't need to check
+        // any signatures now, we will check them for each transaction independently
+        if (method != this.executeTransactions.selector) {
+            address[] memory signers = extractMetaTxSigners(wallet, method, _data);
+            metaTxHash.verifySignatures(signers, signatures);
+        } else {
+            // As `metaTxHash` is the hash of `data` it can't contain the hash of `data`, so set it here
+            require(metaTxHash == 0x0, "INVALID_DATA");
+            assembly {
+                mstore(add(_data, 68), metaTxHash)
```

Comment:
why save the hash, seems it is not used anywhere.

Reply:
does it really just replace the value of the `metaTxHash` parameter in `executeTransactions `?

Reply:
It's an input parameter for `executeTransactions`, so the calldata here is modified directly so the `call` can be done as usual. That's where the signatures are currently checked so we need to know the hash of the complete meta transaction.

I want to make as few changes to `executeMetaTx` as possible, so I'm trying to find a better way to do this.

Reply:
This is actually fine.